src/components/ec-input-field/ec-input-field.spec.ts

```diff
@@ -152,36 +158,35 @@ describe('EcInputField', () => {
       },
     );
 
-    expect(wrapper.findByDataTest('ec-input-field__input').element.value).toBe('');
-    wrapper.findByDataTest('ec-input-field__input').setValue('some text');
-    expect(wrapper.vm.text).toBe('some text');
-    expect(wrapper.findByDataTest('ec-input-field__input').element.value).toBe('some text');
+    expect((findByDataTest(wrapper, 'ec-input-field__input').element as HTMLInputElement).value).toBe('');
+    await findByDataTest(wrapper, 'ec-input-field__input').setValue('some text');
+    expect(wrapper.findComponent(EcInputField).emitted()[InputFieldEvent.UPDATE_MODEL_VALUE]?.[0]).toEqual(['some text']);
+    expect((findByDataTest(wrapper, 'ec-input-field__input').element as HTMLInputElement).value).toBe('some text');
   });
 
   it('should emit an event when we click on the icon', () => {
     const wrapper = mountInputField({ icon: 'simple-check' });
 
-    wrapper
-      .findByDataTest('ec-input-field__icon')
+    findByDataTest(wrapper, 'ec-input-field__icon')
       .trigger('click');
 
-    expect(wrapper.emitted('icon-click').length).toBe(1);
+    expect(wrapper.emitted(InputFieldEvent.ICON_CLICK)?.length).toBe(1);
   });
 
   it('should render given icon', () => {
     const wrapper = mountInputField({ icon: 'simple-check' });
-    expect(wrapper.findByDataTest('ec-input-field__icon-wrapper').exists()).toBe(true);
+    expect(findByDataTest(wrapper, 'ec-input-field__icon-wrapper').exists()).toBe(true);
```

Comment:
It will be necessary only at the moment we rewrite spec file to `ts`. And it is just 2 seconds with replace functionality of IDE. On the other hand this is much more easier to maintain types of pretty simple function then some prototype mixin.

Reply:
To me, the developer experience and code readability goes down. You cannot chain two of these calls anymore `wrapper.findByDataTest('ec-input-field').findByDataTest('ec-input-field__icon-wrapper')` and overall this function doesn't fit to the API that is set by the VTU:

```js
await wrapper.trigger('click');
await wrapper.find('button').trigger('click');
await wrapper.findByDataTest('ec-button').trigger('click');
await wrapper.setValue('test')
```

vs

```js
await wrapper.trigger('click');
await wrapper.find('button').trigger('click');
await findByDataTest(wrapper, 'ec-button').trigger('click');
await wrapper.setValue('test');
```

That function looks now as an alien, that it doesn't belong there. `findByDataTest()` was created as a sister function for `.find()`, it's expected to be called the same way.

I'd rather find a way how to declare the existing function to VueWrapper that changing the code.

Reply:
Okay :)